File: backend/src/main/java/com/HMSync/staff/entity/Staff.java
package com.HMSync.staff.entity;

import com.HMSync.catalog.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "staff")
@Accessors(chain = true)
@EqualsAndHashCode(callSuper = false)
public class Staff extends BaseEntity {
    @Column(name = "first_name")
    private String firstName;

    @Column(name = "last_name")
    private String lastName;

    @Column(name = "phone_number")
    private String phoneNumber;

    @Column(name = "job_title")
    private String jobTitle;

    @Column(name = "department")
    private String department;

    @Column(name = "shift_type")
    private String shiftType;

    @Column(name = "clock_in_time")
    private LocalDateTime clockInTime;

    @Column(name = "clock_out_time")
    private LocalDateTime clockOutTime;

    @Column(name = "performance_review")
    private String performanceReview;

    @Column(name = "assignment")
    private String assignment;
}

File: frontend/HMSync/src/routes/Staff/Staff.js
import { useOutletContext } from "react-router-dom";
import Forms from "../../Forms/Forms";

function Staff() {
    const apiPrefix = useOutletContext();
    const identifier = 'staffId';
    const label = 'firstName';
    const suffixUrl = `${apiPrefix}YDHfCIizOy0f/data`;
    const qKey = 'StaffData';
    const topInfo = {
        'First Name': 'firstName',
        'Staff Id': 'staffId'
    };
    const options = {
        'First Name': 'firstName',
        'Last Name': 'lastName',
        'Staff Id': 'staffId',
        'Phone Number': 'phoneNumber',
        'Job Title': 'jobTitle',
        'Department': 'department',
        'Shift Type': 'shiftType',
        'Clock-in': 'clockInTime',
        'Clock-out': 'clockOutTime',
        'Performance Review': 'performanceReview',
        'Assignment': 'assignment'
    };

    return (
        <Forms
            identifier={identifier}
            topInfo={topInfo}
            options={options}
            label={label}
            apiUrl={suffixUrl}
            qKey={qKey}
        />
    );
}

export default Staff;